tests/integration/components/pix-radio-button-test.js

```diff
@@ -83,6 +83,48 @@ module('Integration | Component | pix-radio-button', function (hooks) {
       assert.false(radiobutton.checked, "Radiobutton has changed state, but shouldn't have");
     });
 
+    test(`it should read success state info if given`, async function (assert) {
```

Comment:
Tester le fait que ça n'ajoute pas l'état si `isDisabled` n'est pas renseigné ?

Reply:
On propose de mettre un warning non visible en prod, comme quand on ne passe pas un booléen à `isDisabled`.

Reply:
Yes nickel 👌

Reply:
C'est ajouté dans un nouveau [commit](ce4c6c84d801e8550daf2f3b53d8ee0a5c3bceb7).

Mais il y a un comportement qui me laisse perplexe dans le test, quand je set l'attribut `@isDisabled` à `false` (booleen) il casse et pas quand je mets `'false'` (string) 🧐

Reply:
Parce que [dans cette ligne de code](https://github.com/1024pix/pix-ui/blob/dev/addon/components/pix-radio-button.js#L22), si on passe `'false'` en string, alors `Boolean("false") === true`.
Et on a donc ce get qui retourne `true`.

Reply:
Oh d'accord merci 💡 Je vais tester ça !

Reply:
En fait, le getter isDisabled ne retournait pas un boolean ce qui faisait péter la condition du warn (celui du state), parce que la ternaire retournait null.

- On avait ça de base : `return this.isDisabled && (!this.hasErrorState || !this.hasSuccessState)`
- Et on a l'a passé comme ça : `return Boolean(this.isDisabled) && (!this.hasErrorState || !this.hasSuccessState)`

On a pas touché au return du isDisabled, pour éviter de casser son comportement initial, mais on a ajouté plusieurs tests unitaires pour s'assurer de bien comprendre ce que ce getter est censé retourner.

Reply:
C'est un peu dangereux parce que `Boolean("false") = true`

Reply:
isDisabled c'est un getter, il renvoie soit `'true'` soit `null`, donc on couvert pour ce cas non ?

Reply:
Non, parce que si la personne a fait `<PixRadioButton @isDisabled="false" />`, alors tu auras ton getter `isDIsabled` qui renverra `true`.

Reply:
alors oui, c'est pas hyper cool pour la DX, mais en l'occurence, si on veut passer false à la props `@isDisabled`, on est censé écrire `<PixRadioButton @isDisabled{{false}} />`

Reply:
Je proposais la soluce pour répondre à la problématique du test : https://github.com/1024pix/pix-ui/pull/679#discussion_r1649122941

Reply:
c'est intéressant, car ce qu'on voulait faire avec Rébecca c'était un test où on passer false en _boolean_, et dans ce cas ça ne fonctionnait pas. Et donc oui en passant `"false"` en string le test fonctionnait.
Hors, on voulait que ça marche avec un _boolean_. Cela nous a amené à revoir la ternaire qu'on utilisait dans l'appel de `Ember.warn`.
Maintenant tu soulèves un point intéressant : quel comportement voulons-nous si on passe bien `false` en _string_ ?
Soit on souhaite qu'un `"false"` soit considéré comme un _boolean_ (et donc `@isDisabled="false"` **ne rend pas** le radio-button _disabled_), soit on décide qu'une chaîne de caractère donne le même résultat que si on la passait à l'attribut natif `disabled` (et donc `@isDisabled="false"` **rend** le radio-button _disabled_)

Reply:
On peut garder le fonctionnement actuel pour l'instant.
Tant que ça marche comme souhaité !

Reply:
Je pense qu'on est bon à l'instant T :) cc @reibecca pour confirmation